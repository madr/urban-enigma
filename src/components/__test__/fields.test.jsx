import React from "react";
import renderer from "react-test-renderer";
import {
  ExerciseField,
  NameField,
  RatingField,
  RepsField,
  WeightField,
  IsWarmupField,
} from "../fields";

describe("ExerciseField", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ExerciseField
          value={"Bänkpress"}
          choices={["Bänkpress", "Marklyft", "Knäböj"]}
          handleOnChange={() => { }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("NameField", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <NameField value={"Bänkpress Boogie 1.1"} handleOnChange={() => { }} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("RatingField", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<RatingField value={3} handleOnChange={() => { }} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("RepsField", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<RepsField value={5} handleOnChange={() => { }} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("WeightField", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<WeightField value={100} handleOnChange={() => { }} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("IsWarmupField", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <IsWarmupField value={true} handleOnChange={() => { }} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});