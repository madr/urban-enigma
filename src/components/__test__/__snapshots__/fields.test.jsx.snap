// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseField renders correctly 1`] = `
Array [
  <datalist
    id="exercises"
  >
    <option>
      Bänkpress
    </option>
    <option>
      Marklyft
    </option>
    <option>
      Knäböj
    </option>
  </datalist>,
  <label
    htmlFor="exercise"
  >
    Övning
  </label>,
  <input
    autoComplete="off"
    id="exercise"
    list="exercises"
    onChange={[Function]}
    value="Bänkpress"
  />,
]
`;

exports[`IsWarmupField renders correctly 1`] = `
<label
  className="label--checkbox"
  htmlFor="isWarmup"
>
  <input
    checked={true}
    id="isWarmup"
    onChange={[Function]}
    type="checkbox"
    value={false}
  />
  Uppvärmningsset
</label>
`;

exports[`NameField renders correctly 1`] = `
Array [
  <label
    htmlFor="name"
  >
    Döp passet
  </label>,
  <input
    id="name"
    onChange={[Function]}
    type="text"
    value="Bänkpress Boogie 1.1"
  />,
]
`;

exports[`RatingField renders correctly 1`] = `
Array [
  <label
    htmlFor="rating"
  >
    Hur gick det?
  </label>,
  <input
    id="rating"
    inputMode="numeric"
    max={5}
    min={1}
    onChange={[Function]}
    pattern="[0-9]*"
    step={1}
    type="range"
    value={3}
  />,
]
`;

exports[`RepsField renders correctly 1`] = `
Array [
  <label
    htmlFor="reps"
  >
    Repetitioner
  </label>,
  <input
    id="reps"
    inputMode="numeric"
    min="1"
    onChange={[Function]}
    pattern="[0-9]*"
    step="1"
    type="number"
    value={5}
  />,
]
`;

exports[`WeightField renders correctly 1`] = `
Array [
  <label
    htmlFor="weight"
  >
    Vikt
  </label>,
  <input
    id="weight"
    inputMode="decimal"
    lang="en-150"
    min="2.5"
    onChange={[Function]}
    pattern="[0-9]*"
    step="0.5"
    type="number"
    value={100}
  />,
]
`;
